spaceship/lib/spaceship/tunes/tunes_client.rb

```diff
@@ -1582,9 +1574,12 @@ def clear_user_cached_data
       @sso_token_for_video = nil
     end
 
-    # the contentProviderIr found in the UserDetail instance
+    # the contentProviderId found in the user details data
     def content_provider_id
-      @content_provider_id ||= user_detail_data.content_provider_id
+      provider = user_details_data["provider"]["providerId"]
+      if provider
+        @content_provider_id ||= provider.to_s
```

Comment:
AFAICT, this method doesn't use the memoized value.

Reply:
Do you mean like "it doesn't get used as the return value"? 🤔 

If so, the last line in the if statement becomes the return value in the if statement which will implicitly be the return value of the method but...

I also don't love how this method looks. I'll clean it up!